starport/services/chain/build.go

```diff
@@ -169,6 +169,22 @@ func (c *Chain) buildProto(ctx context.Context) error {
 			),
 		)
 	}
+
+	// generate ts client code as well if it is enabled.
+	// NB: Vuex generates JS client code as well but within a vuex store.
+	if conf.Client.Typescript.Path != "" {
```

Comment:
Hey, not following here, what logic are we reusing? And what change do we need to replicate specifically?

Reply:
> Inside this block, can we reuse the logic from above?

Basically we have duplicated logic between these two:

https://github.com/tendermint/starport/blob/9b11250d229e1b6dfe6406dce2af9a81aeed7eb5/starport/services/chain/build.go#L160-L170

https://github.com/tendermint/starport/blob/9b11250d229e1b6dfe6406dce2af9a81aeed7eb5/starport/services/chain/build.go#L177-L188

It would be better to avoid duplicating code and making the common logic reusable between two places.

> We also need to make the same change in here.

We do code generation both in the `scaffolder` pkg (when using app, type, etc. commands) and `chain` pkg (when using serve, build commands).

This is why we need to implement the same thing in the `scaffolder` pkg.

We can also make the code generation logic between these two package reusable by moving the logic to a common place and using it in both places, but I think this can be handled by another PR.

Reply:
Ah, yea makes sense I didn't read the filename initially on the link you shared :man_facepalming: .

Refactored the logic into common location :+1: .